classes/form/section_select_form.php

```diff
@@ -87,6 +88,24 @@ public function definition() {
             $radioarray[] = $mform->createElement('radio', 'targetsectionnum',
                 '', $sectionname, $sectionnum, ['class' => 'mt-2']);
         }
+
+        $course = $DB->get_record('course', array('id' => $targetcourseid), '*', MUST_EXIST);
+        $courseformat = course_get_format($course);
+        $courseformatoptions = $courseformat->get_format_options();
+        $maxsections = $courseformat->get_max_sections();
+        $desirednumsections = 0;
+
+        if (isset($courseformatoptions['numsections'])) { // Course format setting.
+            $desirednumsections = $courseformatoptions['numsections'] + 1;
+        } else if (course_get_format($course)->uses_sections()) { // No format setting.
```

Comment:
Any reason for not reusing `$courseformat` here?

Reply:
good catch!